### PR TITLE
Automatic port allocation for unrooted phones compatibility

### PR DESCRIPTION
**Pull request by irinil**
With this addition, protocols with ports <1024 can work on unrooted phones.

### CODE REVIEW

**Comment by sastry17** [comment]
The issue is not done. The app shows that the services were started but Nmap scan discovers only MySQL and SIP open. The services on the device have not started. 

image

**Comment by sastry17** [comment]
Check https://github.com/aau-network-security/HosTaGe/blob/master/assets/payload/redirect-ports.sh

**Comment by irinil** [comment]
Ok I will check the issue!

**Comment by irinil** [comment]
Hello @sastry17 ,

I run an intense scan too, although I used the "all tcp ports argument", because the regular intense scan in an nmap, finds only the known tcp ports. **With my scan as you can see below, all the open ports in the device are displayed** :



**I performed multiple attacks using the discovered ports:
The first example is with the port:45187 which is the Echo protocol.**

2020-06-18 21_03_21-Movies   TV

**Below you can see the recorded result:**

104418118_2102263286583848_5356775761079996520_n

**I tested all the ports, some of them didn't respond (missing http request support), although they where all open.**

104418118_2102263286583848_5356775761079996520_n

104476776_315491169852954_5312070151069036777_n

**Here is my network info:**

104475877_926527094441602_1010585972931281541_n


### Conclusion
**_In conclusion, I think the issue is done. Of course you can check it yourself, with the same Nmap test scan configuration an send me feedback :)_**
